torch/csrc/distributed/rpc/process_group_agent.h

```diff
@@ -12,6 +12,10 @@ namespace torch {
 namespace distributed {
 namespace rpc {
 
+constexpr int kDefaultNumSendRecvThreads = 4;
+constexpr std::chrono::milliseconds kDefaultRpcTimeout =
+    std::chrono::seconds(10);
```

Comment:
distributed autograd, rref and other internal RPCs would end up using this timeout as well. Lets make the default 60s to be conservative. Ideally, this timeout should only affect user RPCs and not system RPCs.

We can do this in future PRs:

1. Have a `timeout` parameter in RpcAgent.send().
2. For user RPCs, we always fill that in with the default rpc timeout.
3. For system RPCs, it'll default to 0 (which would be infinite) unless the system RPC sets it.
4. Later we can allow the user level RPC API to specify timeouts as well.

Reply:
Created https://github.com/pytorch/pytorch/issues/29018 to track this.